src/parser.ts

```diff
@@ -785,19 +792,20 @@ export class Parser {
         return this.finalize(node, new Node.FunctionExpression(null, params.params, method, isGenerator));
     }
 
-    parsePropertyMethodAsyncFunction(): Node.FunctionExpression {
+    parsePropertyMethodAsyncFunction(isGenerator: boolean): Node.FunctionExpression {
```

Comment:
Might make sense at some point to put isGenerator into context

Reply:
This is [Boolean trap](https://ariya.io/2011/08/hall-of-api-shame-boolean-trap), though we already have some cases like that in the same file.

Reply:
So should I put this into `context` as part of this PR or what changes required?